test/Microsoft.DotNet.Cli.Utils.Tests/DangerousFileDetectorTests.cs

```diff
@@ -0,0 +1,109 @@
+{
+    public class DangerousFileDetectorTests : TestBase
+    {
+        [WindowsOnlyFact]
+        public void ItShouldDetectFileWithMarkOfTheWeb()
+        {
+            var testFile = Path.Combine(TempRoot.Root, Path.GetRandomFileName());
+            File.WriteAllText(testFile, string.Empty);
+            AlternateStream.WriteAlternateStream(
+                testFile,
+                "Zone.Identifier",
+                "[ZoneTransfer]\r\nZoneId=3\r\nReferrerUrl=C:\\Users\\test.zip\r\n");
+
+            new DangerousFileDetector().IsDangerous(testFile).Should().BeTrue();
+        }
+
+        [Fact]
+        public void WhenThereIsNoFileItReturnsFalse()
+        {
+            var testFile = Path.Combine(TempRoot.Root, Path.GetRandomFileName());
+
+            new DangerousFileDetector().IsDangerous(testFile).Should().BeFalse();
+        }
+
+        [NonWindowsOnlyFact]
+        public void WhenRunOnNonWindowsReturnFalse()
+        {
+            var testFile = Path.Combine(TempRoot.Root, Path.GetRandomFileName());
+            File.WriteAllText(testFile, string.Empty);
+
+            new DangerousFileDetector().IsDangerous(testFile).Should().BeFalse();
+        }
+
+        private static class AlternateStream
+        {
+            private const uint GenericWrite = 0x40000000;
+
+            public static void WriteAlternateStream(string filePath, string altStreamName, string content)
+            {
+                if (string.IsNullOrWhiteSpace(filePath))
+                {
+                    throw new ArgumentException("message", nameof(filePath));
+                }
+
+                if (string.IsNullOrWhiteSpace(altStreamName))
+                {
+                    throw new ArgumentException("message", nameof(altStreamName));
+                }
+
+                if (content == null)
+                {
+                    throw new ArgumentNullException(nameof(content));
+                }
+
+                string altStream = filePath + ":" + altStreamName;
+
+                SafeFileHandle fileHandle
+                    = CreateFile(
+                        filename: altStream,
+                        desiredAccess: GenericWrite,
+                        shareMode: 0,
+                        attributes: IntPtr.Zero,
+                        creationDisposition: (uint)FileMode.CreateNew,
+                        flagsAndAttributes: 0,
+                        templateFile: IntPtr.Zero);
+
+                if (!fileHandle.IsInvalid)
+                {
+                    using (var streamWriter = new StreamWriter(new FileStream(fileHandle, FileAccess.Write)))
+                    {
+                        streamWriter.WriteLine(content);
+                        streamWriter.Flush();
+                    }
+                }
+                else
+                {
+                    Exception ex = new Win32Exception(Marshal.GetLastWin32Error());
+                    if (!ex.Message.Contains("cannot find the file"))
```

Comment:
We should check the numeric error code here. I think this could cause the test to fail on non-English OS.

Reply:
forget to correct this case. but since it is test we should just it throw without checking the message. the error should give the same info

Reply:
In the end I changed to this. No special check for cannot find file. But I still need throw ex

```c#
if (!fileHandle.IsInvalid)
                {
                    using (var streamWriter = new StreamWriter(new FileStream(fileHandle, FileAccess.Write)))
                    {
                        streamWriter.WriteLine(content);
                        streamWriter.Flush();
                    }
                }
                else
                {
                    Exception ex = new Win32Exception(Marshal.GetLastWin32Error());
                    throw ex;
                }
```

Reply:
OK. Sorry, I'm so used to reading throw ex as a mistaken rethrow. Any reason not to write this as `throw new Win32Exception(...)`. It's more idiomatic.

Reply:
No special reason, I changed it to `throw new Win32Exception(Marshal.GetLastWin32Error());`